gtsam/discrete/TableFactor.cpp

```diff
@@ -254,6 +254,22 @@ DecisionTreeFactor TableFactor::operator*(const DecisionTreeFactor& f) const {
   return toDecisionTreeFactor() * f;
 }
 
+/* ************************************************************************ */
+DiscreteFactor::shared_ptr TableFactor::multiply(
+    const DiscreteFactor::shared_ptr& f) const {
+  DiscreteFactor::shared_ptr result;
+  if (auto tf = std::dynamic_pointer_cast<TableFactor>(f)) {
+    result = std::make_shared<TableFactor>(this->operator*(*tf));
```

Comment:
I would think operator* with call multiply and not the other way around

Reply:
But then operator* would have to accept a shared pointer and your previous comment contradicts this.

Reply:
You're right.